include/seqan3/range/view/pairwise_combine.hpp

```diff
@@ -0,0 +1,595 @@
+        constexpr iterator_type operator+(difference_type const offset)
+            noexcept(noexcept(std::declval<iterator_type &>() += 1))
+        //!\cond
+            requires std::RandomAccessIterator<source_iterator_type>
+        //!\endcond
+        {
+            iterator_type tmp{*this};
+            return (tmp += offset);
+        }
+
+        //!\brief Advances the iterator by the given offset; `source_iterator_type` must model std::RandomAccessIterator.
+        constexpr friend iterator_type operator+(difference_type const offset, iterator_type iter)
+            noexcept(noexcept(std::declval<iterator_type &>().from_index(1)))
+        //!\cond
+            requires std::RandomAccessIterator<source_iterator_type>
+        //!\endcond
+        {
+            iter.from_index(iter.to_index() + offset);
+            return iter;
+        }
+
+        //!\brief Decrements the iterator by the given offset; `source_iterator_type` must model std::RandomAccessIterator.
+        constexpr iterator_type & operator-=(difference_type const offset)
+            noexcept(noexcept(std::declval<iterator_type &>().from_index(1)))
+        //!\cond
+            requires std::RandomAccessIterator<source_iterator_type>
+        //!\endcond
+        {
+            from_index(to_index() - offset);
+            return *this;
+        }
+
+        //!\brief Decrements the iterator by the given offset; `source_iterator_type` must model std::RandomAccessIterator.
+        constexpr iterator_type operator-(difference_type const offset)
+            noexcept(noexcept(std::declval<iterator_type &>() -= 1))
+        //!\cond
+            requires std::RandomAccessIterator<source_iterator_type>
+        //!\endcond
+        {
+            iterator_type tmp{*this};
+            return (tmp -= offset);
+        }
+
+        //!\brief Computes the distance between two iterators; `source_iterator_type` must model std::RandomAccessIterator.
+        constexpr friend difference_type operator-(iterator_type const & lhs, iterator_type const & rhs)
+            noexcept(noexcept(std::declval<iterator_type &>().to_index()))
+        //!\cond
+            requires std::RandomAccessIterator<source_iterator_type>
+        //!\endcond
+        {
+            return static_cast<difference_type>(lhs.to_index() - rhs.to_index());
+        }
+        //!\}
+
+        /*!\name Comparison operators
+         * \brief These operators are available if the `source_iterator_type` models std::EqualityComparable or
+         *        std::StrictTotallyOrdered respectively.
+         * \{
+         */
+        constexpr friend bool operator==(iterator_type const & lhs, iterator_type const & rhs)
+            noexcept(noexcept(std::declval<source_iterator_type &>() == std::declval<source_iterator_type &>()))
+        //!\cond
+            requires std::EqualityComparable<source_iterator_type>
```

Comment:
that is unfortunately true and I did not thought about it when implementing the operators. But since the operators constraints are not participating in any overload resolution it works without the constraints as well. Only the emitted compiler error will be not as expected until the bug is resolved. So I tend to keep it this way as it is the suggested way to implement the operators.